core/src/main/java/org/apache/iceberg/deletes/TargetedPositionDeleteWriter.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.deletes;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.function.Supplier;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.io.DeleteWriteResult;
+import org.apache.iceberg.io.FileWriter;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.util.CharSequenceSet;
+
+/**
+ * A position delete writer that produces a separate delete file for each referenced data file.
+ *
+ * <p>This writer does not keep track of seen deletes and assumes all incoming records are ordered
+ * by file and position as required by the spec. If there is no external process to order the
+ * records, consider using {@link SortingPositionOnlyDeleteWriter} instead.
+ */
+public class TargetedPositionDeleteWriter<T>
```

Comment:
DataFileTargetedPositionDeleteWriter?

Reply:
I tend to prefer shorter names if possible given our new 100 line length limit. Do you think this name will be easier to understand?

Reply:
I guess the naming needs to somehow reflect the granularity which will make things more clear.

Reply:
I agree, let me think a bit more about this. If you have more ideas, please share them as well!

Reply:
`SingleFilePosDeleteWriter`?

Reply:
I believe `Clustered` is something we use for `PartitioningWriter` implementations to indicate that incoming records are grouped by spec and partition. If we use that prefix in this context, it may be a bit misleading.

I renamed this class to `FileScopedPositionDeleteWriter`. Let me know what you think.

Reply:
I think FileScoped or if you want a whole new name PerFilePostionDeleteFileWriter?